.pipelines/e2e.yml

```diff
@@ -10,47 +10,94 @@ resources:
 # Azure DevOps Pipeline running e2e tests
 variables:
 - template: vars.yml
-jobs:
-- job: E2E
-  timeoutInMinutes: 180
-  pool:
-    name: ARO-CI
-    demands: go-1.16
-  steps:
-  - template: ./templates/template-checkout.yml
-  - template: ./templates/template-az-cli-login.yml
-    parameters:
-      azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
-  - script: |
-      az account set -s $AZURE_SUBSCRIPTION_ID
-      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
-      make secrets
-      . secrets/env
-      echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
-    displayName: 🔑 Downloading certificates and secrets from storage account
-    name: setEnv
-  - template: ./templates/template-push-images-to-acr.yml
-    parameters:
-      rpImageACR: $(RP_IMAGE_ACR)
-      buildCommand: publish-image-aro
-  - script: |
-      set -e
-      set -o pipefail
-
-      . secrets/env
-
-      az account set -s $AZURE_SUBSCRIPTION_ID
-
-      set -x
-      . ./hack/e2e/run-rp-and-e2e.sh
-      trap 'set +e; kill_rp; clean_e2e_db' EXIT
-
-      deploy_e2e_db
-
-      run_rp
-      validate_rp_running
-      register_sub
-
-      export CI=true
-      make test-e2e
-  - template: ./templates/template-az-cli-logout.yml
+
+# Run the test suite
+stages:
+- stage: A
```

Comment:
can these stage names be anything we want?
A -> "execute-tests"
B -> "published-failed-must-gather"
C -> "clean-up"

Reply:
@s-amann that was my thought too, but unfortunately I wasn't able to get it to work when the stages had names other than A, B, and C. This is partially addressed by having distinct job names, but agreed its not ideal.

Reply:
It _should_ work, maybe quotes are needed to properly parse the string of a better named stage?

Reply:
It works, it just doesn't like spaces or quotes :)

Let me know if the naming makes sense @s-amann @bryanro92. The collect must-gather stage needs some work still, looks like.

Reply:
I like the names